File: src/dataset/c/space_age_Exercism_chriswilding/gemini/GeminiFlash_space_age_v2.c
#include "space_age.h"

static const float orbital_period_factors[] = {
    seconds_in_earth_on_year * 0.2408467f,   // Mercury
    seconds_in_earth_on_year * 0.61519726f,  // Venus
    seconds_in_earth_on_year * 1.0f,          // Earth
    seconds_in_earth_on_year * 1.8808158f,   // Mars
    seconds_in_earth_on_year * 11.862615f,  // Jupiter
    seconds_in_earth_on_year * 29.447498f,  // Saturn
    seconds_in_earth_on_year * 84.016846f,  // Uranus
    seconds_in_earth_on_year * 164.79132f   // Neptune
};

float convert_planet_age(planet_t planet, int64_t input) {
  return (float)input / orbital_period_factors[planet];
}

File: src/dataset/typescript/space-age_Exercism-typescript-alemarr/gemini/GeminiFlash_space-age_v2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.age = void 0;
const SECONDS_IN_YEAR = 31557600;
const orbitalPeriods = {
    earth: 1,
    mercury: 0.2408467,
    venus: 0.61519726,
    mars: 1.8808158,
    jupiter: 11.862615,
    saturn: 29.447498,
    uranus: 84.016846,
    neptune: 164.79132
};
const planetYears = Object.fromEntries(Object.entries(orbitalPeriods).map(([planet, period]) => [planet, period * SECONDS_IN_YEAR]));
const age = (planet, seconds) => {
    const planetSecond = planetYears[planet];
    return Math.round((seconds / planetSecond) * 100) / 100;
};
exports.age = age;
